src/StoreApi/Utilities/ProductQueryFilters.php

```diff
@@ -211,4 +211,58 @@ public function get_rating_counts( $request ) {
 
 		return array_map( 'absint', wp_list_pluck( $results, 'product_count', 'rounded_average_rating' ) );
 	}
+
+	/**
+	 * Gets product by filtered terms.
+	 *
+	 * @since TBD
+	 * @param string $taxonomy Taxonomy name.
+	 * @param array  $term_ids Term IDs.
+	 * @param string $query_type or | and.
+	 * @return array Product IDs.
+	 */
+	public function get_product_by_filtered_terms( $taxonomy = '', $term_ids = array(), $query_type = 'or' ) {
+		global $wpdb;
+
+		$term_count = count( $term_ids );
+		$results    = array();
+		$term_ids   = implode( ',', array_map( 'intval', $term_ids ) );
+
+		if ( 'or' === $query_type ) {
+			// phpcs:disable
+			$results = $wpdb->get_col(
+				$wpdb->prepare(
+					"
+					SELECT DISTINCT `product_or_parent_id`
+					FROM {$wpdb->prefix}wc_product_attributes_lookup
+					WHERE `taxonomy` = %s
+					AND `term_id` IN ({$term_ids})
```

Comment:
We can also `$wpdb->prepare` the `$term_ids` here:
```suggestion
					AND `term_id` IN (%s)",
```

Reply:
As a second thought: we can keep as-is, since we are already running `intval` on this variable.